TMDS_3/include/CGAL/tetrahedron_soup_to_triangulation_3.h

```diff
@@ -0,0 +1,181 @@
+//
+//
+// Author(s)     :  Mael Rouxel-Labbé, Maxime Gimeno, Jane Tournois
+//
+//******************************************************************************
+// File Description :
+//******************************************************************************
+
+#ifndef CGAL_TMDS_3_TETRAHEDRON_SOUP_TO_C3T3_H
+#define CGAL_TMDS_3_TETRAHEDRON_SOUP_TO_C3T3_H
+
+#include <CGAL/license/TMDS_3.h>
+
+#include <CGAL/TMDS_3/tet_soup_to_c3t3.h>
+#include <CGAL/TMDS_3/internal/TMDS_3_helper.h>
+#include <CGAL/Named_function_parameters.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+
+#include <vector>
+#include <array>
+#include <boost/unordered_map.hpp>
+
+namespace CGAL {
+
+  /** \ingroup PkgTMDS3Functions
+  * builds a 3D triangulation from a soup of tetrahedra.
+  *
+  * @tparam TetrahedronRange a model of `Range` whose value type is
+  * a `Tetrahedron_3`. The point type of the tetrahedra must be convertible
+  * to `Triangulation::Point`
+  * @tparam Triangulation a 3D triangulation class that has
+  * a vertex base model of `MeshVertexBase_3`
+  * and a cell base model of `MeshCellBase_3`
+  *
+  * @param tets the set of finite tetrahedra of a valid CGAL triangulation.
+  * Each element in the range is the geometric description of the
+  * corresponding cell in `tr`
+  *
+  * @returns the 3D triangulation built from \p tets
+  *
+  * @post the output triangulation must be a triangulation of the convex hull of `tets`
+  */
+  template<typename Triangulation, typename TetrahedronRange>
+  Triangulation tetrahedron_soup_to_triangulation_3(const TetrahedronRange& tets)
+  {
+    using Tr    = Triangulation;
+    using Point = typename Tr::Point;
+
+    Triangulation tr;
+    std::vector<Point> points;
+    std::vector<std::array<int, 5> > finite_cells;
+    boost::unordered_map<std::array<int, 3>, typename Tr::Cell::Surface_patch_index> border_facets;
+    boost::unordered_map<Point, int> p2i;
+
+    CGAL_assertion_code(
+      typename Triangulation::Geom_traits::Orientation_3 orientation =
+      tr.geom_traits().orientation_3_object();
+    );
+
+    for (typename TetrahedronRange::value_type tet : tets)
+    {
+      CGAL_assertion(tet.orientation() == CGAL::POSITIVE);
+      std::array<int, 5> cell;
+
+      for (int i = 0; i < 4; ++i)
+      {
+        const Point& pi = tet[i];
+        if (p2i.find(pi) == p2i.end())
+        {
+          points.push_back(pi);
+          int index = static_cast<int>(points.size() - 1);
+          p2i.insert(std::make_pair(pi, index));
+          cell[i] = index;
+        }
+        else
+          cell[i] = p2i.at(pi);
+      }
+      cell[4] = 1;
+
+      CGAL_assertion(orientation(points[cell[0]],
+        points[cell[1]], points[cell[2]], points[cell[3]]) == CGAL::POSITIVE);
+
+      finite_cells.push_back(cell);
+    }
+
+    CGAL::TMDS_3::build_triangulation(tr, points, finite_cells, border_facets);
+
+    CGAL_assertion(CGAL::TMDS_3::internal::is_convex(tr));
+
+    return std::move(tr);
```

Comment:
```suggestion
    return tr;
```

Just return by copy. The use of `std::move` prevent the copy-elision.

See https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-return-move-local or the more detailed http://vmpstr.blogspot.com/2015/12/redundant-stdmove.html.